### PR TITLE
Switch to use parser library

### LINKED ISSUE #4117: Switch to use the parser from devfile/parser in odo repo
After https://github.com/devfile/api/issues/183 is completed, we want to switch to use the parser from devfile/parser repo, and remove the parser defined in Odo. 

Since the parser library is using devfile go struct defined in devfile/api, this requires some code changes to Odo code base to integrate with the new devfile go struct.  product code and test code changes are expected. 

**Note: the validate package will temporarily be left in odo, the validator will be implemented in devfile/api repo. 
For any call to the following parser functions in odo, need to call `validate.ValidateDevfileData` after, separately: 
 `ParseAndValidate`, `ParseFromURLAndValidate`, `ParseFromDataAndValidate`,

### PR DESCRIPTION
**Pull request by yangcao77**
**What type of PR is this?**

/area devfile
/kind code-refactoring

**What does does this PR do / why we need it**:
This PR removes the parser pkg from odo repo, and use parser from devfile/library instead
This PR also integrate odo with the devfile go structs defined in devfile/api

**Which issue(s) this PR fixes**:

Fixes #4117 

**PR acceptance criteria**:

- [x] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:
run integration tests and unit tests to make sure no regression has been introduced

### CODE REVIEW

**Comment by yangcao77** [comment]
This PR is currently blocked by: https://github.com/devfile/api/pull/203
the yaml tags need to be added in structs for devfile writer to work

**Comment by kadel** [comment]
@yangcao77 It looks like there is something broken in headlight the paths (`/` vs `\`)
Windows unit tests are failing on 
```
--- FAIL: TestCopyDirWithFS (0.00s)
    --- FAIL: TestCopyDirWithFS/case_1:_folder_with_a_nested_file_and_folder (0.00s)
        util_test.go:2286: MoveDir() error = chmod C:\Users\travis\AppData\Local\Temp\destination/blah.js: file does not exist, wantErr false
        util_test.go:2340: some folder were not created
        util_test.go:2344: some files were not created
--- FAIL: TestGitSubDir (0.00s)
    --- FAIL: TestGitSubDir/case_1:_normal_sub_dir_exist (0.00s)
        util_test.go:2570: GitSubDir() error = chmod C:\Users\travis\AppData\Local\Temp\destination\java/main.java: file does not exist, wantErr false
        util_test.go:2611: all files were not copied
```

Notice the forward flash instead of back slash in the path.

**Comment by yangcao77** [comment]
@kadel that is wired.. I did not modify those two functions / files. I will rebase with the latest master to see.

**Comment by kadel** [comment]
> This PR is currently blocked by: [devfile/api#203](https://github.com/devfile/api/pull/203)
> the yaml tags need to be added in structs for devfile writer to work

@yangcao77 I see the tests are passing. Is this PR still blocked on https://github.com/devfile/api/pull/203?

**Comment by yangcao77** [comment]
@kadel It's unblocked now. people can start reviewing the PR.

File: pkg/config/config_test.go [hunk 1]
```diff
@@ -518,3 +697,49 @@ func createDirectoryAndFile(create bool, fs filesystem.Filesystem, odoDir string
 	}
 	return nil
 }
+
+func testDevfileObj(fs devfilefs.Filesystem) parser.DevfileObj {
```
Comment (mik-dass) [inline pkg/config/config_test.go:701]: Please move this to the testingutil package

**Comment by kadel** [comment]
Good job @yangcao77 
/approve

File: pkg/config/config_test.go (hunk 1 above)
Comment (yangcao77) [reply pkg/config/config_test.go]: updated

**Review by mik-dass — APPROVED** [review]
CI is green
/lgtm

**Comment by yangcao77** [comment]
@mik-dass Can you help add back the lgtm label? the label  was removed as I did a rebase

**Comment by mik-dass** [comment]
> @mik-dass Can you help add back the lgtm label? the label  was removed as I did a rebase

Sure.
/lgtm

**Pull request merged**